src/main/java/Storage/TaskList.java

```diff
@@ -0,0 +1,126 @@
+package storage;
+
+import taskmanager.Task;
+
+import java.util.ArrayList;
+
+/**
+ * The `TaskList` class behave like an ArrayList
+ * to store a list of tasks.
+ */
+public class TaskList {
+    private ArrayList<Task> task; //The array list to store all the tasks.
+
+    /**
+     * Constructs a `TaskList` with the specified list of tasks.
+     *
+     * @param task The ArrayList containing tasks to initialize the task list.
+     */
+    public TaskList(ArrayList<Task> task) {
+        this.task = task;
+    }
+
+    public TaskList() {
+        this.task = new ArrayList<>();
+    }
+
+    /**
+     * Adds task to task list.
+     *
+     * @param t The task to be added to the list.
+     */
+    public void add(Task t) {
+        task.add(t);
+    }
+
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return The number of tasks in the list.
+     */
+    public int size() {
+        return this.task.size();
+    }
+
+    /**
+     * Checks if the task list is empty.
+     *
+     * @return True if the task list is empty; otherwise, false.
+     */
+    public boolean isEmpty() {
+        return this.task.isEmpty();
+    }
+
+    /**
+     * Removes a task at the specified index from the task list.
+     *
+     * @param i The index of the task to be removed.
+     */
+    public void remove(int i) {this.task.remove(i);}
```

Comment:
can consider using egyptian brackets for consistency
```suggestion
    public void remove(int i) {
        this.task.remove(i);
    }
````